### ISSUE TITLE
Move decomposable StableHLO ops to CHLO

### ISSUE THREAD

**Issue by burmako** [openxla/stablehlo#3]
In https://discourse.llvm.org/t/rfc-proposal-for-a-high-level-ml-dialect-in-mlir/64249/46, we did a survey of MHLO ops (almost of that applies to StableHLO, since we've bootstrapped it from MHLO) and identified the following ops which are decomposable to other MHLO ops: `batch_norm_grad`, `batch_norm_inference`, `batch_norm_training`, `broadcast`, `create_token`, `cross-replica-sum`, `dot`, `einsum`, `torch_index_select`, `trace`, `unary_einsum`. Let's move these ops to CHLO.

Note: In MHLO, `broadcast` and `dot` are referring to less general versions of `broadcast_in_dim` and `dot_general`. In HLO, `broadcast` and `dot` are used to refer to the more general versions, so we have a bit of confusing naming here which this move will also help to solve.

**Comment by burmako** [comment]
In retrospect, this ticket was a bit ahead of its time:
  1) The fact that we haven't established a baseline specification prevented us from effectively planning the move. For example, as discussed above, it turned out that we cannot move `broadcast` to CHLO, because unranked broadcasts cannot be expressed via `broadcast_in_dim`.
  2) Our compatibility guarantees haven't yet been finalized (see #115), so it's unclear what exactly this move will entail. E.g. are we going to keep both `chlo.broadcast` and `stablehlo.broadcast` around? For how long? How do we migrate users?
  3) We don't yet have an RFC process to reason about changes to the StableHLO opset. #196 was only introduced a few days ago and will still need some additional work.

As a result, we'll be focusing on completing the StableHLO v1.0 spec in the next few months, and by then we expect that all the blockers mentioned above will be resolved. Until then, I'll be unassigning this ticket.

**Comment by burmako** [comment]
Refactored this ticket to focus on ops which don't have equivalents in the HLO opset (not counting dynamism ops which are handled separately in #8). BatchNorm ops which were previously part of this ticket are now discussed in #603.
